File: common/src/main/java/com/pluxity/facility/dto/FacilityHistoryResponse.java
package com.pluxity.facility.dto;

import com.pluxity.facility.history.FacilityHistory;
import com.pluxity.file.dto.FileResponse;

public record FacilityHistoryResponse(
        Long id, String comment, String createdAt, String createdBy, FileResponse file) {
    public static FacilityHistoryResponse from(FacilityHistory entity, FileResponse file) {
        return new FacilityHistoryResponse(
                entity.getId(),
                entity.getComment(),
                entity.getCreatedAt().toString(),
                entity.getCreatedBy(),
                file);
    }
}

File: common/src/main/java/com/pluxity/facility/dto/FacilityHistoryResponse.kt
package com.pluxity.facility.dto

import com.pluxity.facility.history.FacilityHistory
import com.pluxity.file.dto.FileResponse

data class FacilityHistoryResponse(
    val id: Long,
    val comment: String,
    val createdAt: String,
    val createdBy: String,
    val file: FileResponse,
)

fun FacilityHistory.toHistoryResponse(file: FileResponse): FacilityHistoryResponse =
    FacilityHistoryResponse(
        id = this.id!!,
        comment = this.comment,
        createdAt = this.createdAt.toString(),
        createdBy = this.createdBy.toString(),
        file = file,
    )
